deepchecks/core/checks.py

```diff
@@ -160,6 +167,62 @@ def metadata(self, with_doc_link: bool = False) -> CheckMetadata:
             summary=get_docs_summary(self, with_doc_link)
         )
 
+    def config(self, with_module=True) -> CheckConfig:
+        """Return check config (conditions' config not yet supported).
+
+        Parameters
+        ----------
+        with_module : bool, default True
+            whethere to include the module path or not
+
+        Returns
+        -------
+        CheckConfig
+            includes the checks class name, params, and module if with_module was True
+        """
+        module_name = self.__module__
+        if module_name.startswith('deepchecks.'):
+            module_name = 'deepchecks.' + module_name.split('.')[1]
+        conf = CheckConfig(
+            class_name=self.__class__.__name__,
+            params=self.params(show_defaults=True),
+        )
+        if with_module:
+            conf['module_name'] = module_name
+        return conf
+
+    @staticmethod
+    def from_config(conf: CheckConfig, module_name: str = None) -> 'BaseCheck':
+        """Return check object from a CheckConfig object.
+
+        Parameters
+        ----------
+        conf : CheckConfig
+            the CheckConfig object
+        module_name: str , default: None
+            if the CheckConfig doesn't include the module name will use this name.
+
+        Returns
+        -------
+        BaseCheck
+            the check class object from given config
+
+        Raises
+        ------
+        DeepchecksValueError
+            if the CheckConfig doesn't include the module name and module_name is None.
+        """
+        module_name = conf.get('module_name', module_name)
+        if module_name is None:
+            raise DeepchecksValueError(
+                'Check Config must include module name or it should be passed to the function.')
+        is_deepchecks = module_name.startswith('deepchecks.')
+        if is_deepchecks:
+            module_name += '.checks'
+        module = importlib.import_module(module_name)
+        return getattr(module, conf['class_name'])(**conf['params'])
```

Comment:
it is worth verifying whether all checks store their init params, IMO, not all do this/

Reply:
wdym store?

Reply:
the params are taken from the init function and I'm calling the init

Reply:
i don't use any param used in the init itself, it is by the signature

Reply:
oh, but the validation needs to be in the param function then

Reply:
if a parameter isn't stored it is a problem we can't fix rn, we relied on this function and the behavior remains the same in my pr

Reply:
1. yes, it is true for user checks, but do our checks store all init params, and under what name? did you check that? that was my initial question)
2. IMO, it is worth modifying `initvars` because you will not get default values from the init signature if init params were not assigned to the objected, this could be changed

Reply:
still not related to this pr particularly, we might want to reconsider hot that function work, but should be a new issue.

Reply:
point 1.

`MultiModelPerformanceReport` stores parameter `alternative_scorers` under name `user_scorers`

```python
    def __init__(self, alternative_scorers: Dict[str, Callable] = None, **kwargs):
        super().__init__(**kwargs)
        self.user_scorers = alternative_scorers
```

you will not get this parameter by calling `initvars`

Reply:
same for `SegmentPerformance`
note: I know that callable serialization is currently not supported by this method, but `alternative_scorer` also could be of type `Dict[str, str]` or `Tuple[str, str]`

Reply:
Also we have parameters which stores default functions which is also problematic no?
Image segment performance
```
        if image_properties:
            self.image_properties = validate_properties(image_properties)
        else:
            self.image_properties = default_image_properties
```

Reply:
still should be a new issue, it is a bug about how we get the params

Reply:
https://github.com/deepchecks/deepchecks/issues/1739